### ISSUE TITLE
"QGDict::hashAsciiKey: Invalid null key" when using anonymous union/struct

### ISSUE THREAD

**Issue by reylau** [doxygen/doxygen#6657]
I am using Doxygen V1.8.14. The following code generates three "QGDict::hashAsciiKey: Invalid null key" errors:

```
/************************************************************************************************************************************************************************************/
/*                                                                                                                                                                                  */
/* @defgroup   Sft        Software Timers.                                                                                                                                          */
/* @brief                 Provides software timers to compensate for the shortage of hardware timers.                                                                               */
/*                                                                                                                                                                                  */
/************************************************************************************************************************************************************************************/

#include <stdint.h>

/** @defgroup SftInt      Module Interface.
 *  @ingroup  Sft
 *  @brief                This describes public module objects, i.e. objects which may be used by the rest of the FW.
 */

/** @defgroup SftICtVar1  1.129. Objects for Managing the Soft Timers.
 *  @ingroup  SftInt
 *  @{
 */

/** @brief Empty comment */
static volatile struct
   {
   union
      {
      struct
         {
         uint8_t : 1;
         }
      structFlag;
      }
   unionRun;

   struct
      {
      uint8_t ui8ConfigurationComplete;
      uint8_t ui8ConfigurationWordReceptionTimeout;
      uint8_t ui8ConfigurationAcknowledgmentRepetitionPeriod;
      }
   structCounter;
   }
            module_vstructSoftTimers;
/** @} */
```

Notice that the union has been simplified and contained useful members in the original code (which runs OK). If I am re-writing the code as follows, I do not get any error:

```
/************************************************************************************************************************************************************************************/
/*                                                                                                                                                                                  */
/* @defgroup   Sft        Software Timers.                                                                                                                                          */
/* @brief                 Provides software timers to compensate for the shortage of hardware timers.                                                                               */
/*                                                                                                                                                                                  */
/************************************************************************************************************************************************************************************/

#include <stdint.h>

/** @defgroup SftInt      Module Interface.
 *  @ingroup  Sft
 *  @brief                This describes public module objects, i.e. objects which may be used by the rest of the FW.
 */

/** @defgroup SftICtVar1  1.129. Objects for Managing the Soft Timers.
 *  @ingroup  SftInt
 *  @{
 */

/** @brief Empty comment */
union unionRun
   {
   struct
      {
      uint8_t : 1;
      }
   structFlag;
   };

/** @brief Empty comment */
struct structCounter
   {
   uint8_t ui8ConfigurationComplete;
   uint8_t ui8ConfigurationWordReceptionTimeout;
   uint8_t ui8ConfigurationAcknowledgmentRepetitionPeriod;
   };

/** @brief Empty comment */
static volatile struct
   {
   union unionRun unionRun;

   struct structCounter structCounter;
   }
            module_vstructSoftTimers;
/** @} */
```

It seems close to #742715 and should have been addressed in version 1.8.11...

If you need more inputs, let me know.

Best regards

L. Rey

**Comment by albert-github** [comment]
I cannot reproduce the problem you have with version 1.8.14. It might be some settings in you Doxyfile.

Can you please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem?

**Comment by reylau** [comment]
Hello albert,

I have joined a self contained example in the zip.

Laurent
[Documentation.zip](https://github.com/doxygen/doxygen/files/2643492/Documentation.zip)

**Comment by albert-github** [comment]
I've run the example with 1.8.13, 1.8.14 and the current master (from the 10... directory) and in all cases I see:

> Searching for tag less structs...
> QGDict::hashAsciiKey: Invalid null key
> QGDict::hashAsciiKey: Invalid null key
> QGDict::hashAsciiKey: Invalid null key

("Nice" those names with the spaces  ;-) )

**Comment by reylau** [comment]
Great that you are able to reproduce the issue!

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6659

**Comment by reylau** [comment]
Thank you!

**Comment by albert-github** [comment]
Code has been integrated in master on github.

**Comment by reylau** [comment]
Great. Thank you. Laurent

⁣Envoyé par TypeApp ​

Le 5 déc. 2018 à 09:40, à 09:40, albert-github <notifications@github.com> a écrit:

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.15.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
